### PR TITLE
chore(release): add release process

### PR DESCRIPTION
**Pull request by vvo**
+ add specific algolia agent to the js client
+ add npm run release, npm run release:beta
+ add CONTRIBUTING guide
+ finish rename to /docsearch

### CODE REVIEW

**Comment by vvo** [comment]
After this one I will do a test release to check the whole process is ok

File: scripts/build-css [hunk 1]
```diff
@@ -2,9 +2,8 @@
 
 set -e # exit when error
 
-VERSION=$(json version < package.json)
-NAME='docsearch.js'
-LICENSE="/*! ${NAME} ${VERSION:-UNRELEASED} | © Algolia | github.com/algolia/docsearch.js */"
+NAME='docsearch'
```
Comment (pixelastic) [inline scripts/build-css]: Why did you remove the `VERSION`?

Comment (vvo) [inline scripts/build-css]: the VERSION is passed by the release script to the build script.

Comment (pixelastic) [inline scripts/build-css]: Can we still manually run `npm run build:css|js` or will it fail?

Comment (vvo) [inline scripts/build-css]: It will work, the version will be "UNRELEASED" in the licence file that's it.

When would you need to do so? (Or just curious?)

Comment (pixelastic) [inline scripts/build-css]: The example in `dev` uses symlinks that points to the css files in `dist`, so I need to run `npm run build:css` when I run `npm run dev`

Comment (vvo) [inline scripts/build-css]: ok it will work yes
